### ISSUE TITLE
Class.alias_method causes private method `alias_method' called for Class (NoMethodError)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1617]
This construct is used in [ERB handler](https://github.com/rails/rails/blob/master/actionview/lib/action_view/template/handlers/erb.rb#L20) in ActionView. And it's public since Ruby 2.5

To reproduce:

```ruby
class Hello
  def world; end

  self.alias_method :earth, :world
end
```

Error:

```
alias_method.rb:4:in `Hello': private method `alias_method' called for Class (NoMethodError)
	from alias_method.rb:1:in `<main>'
```

There is no issue in MRI.

**Comment by deepj** [comment]
@eregon Hello, can you take a look at this since it's the same issue like https://github.com/oracle/truffleruby/commit/19c2b9a81385889e622c552188483d0099267278?

**Comment by eregon** [comment]
OK, will do. I didn't know (or forgot) `Module#alias_method` became public.

A good way to catch this should be our check for the same public methods as MRI:
https://github.com/oracle/truffleruby/blob/master/spec/truffle/methods_spec.rb
https://github.com/oracle/truffleruby/blob/master/spec/tags/truffle/methods_tags.txt

**Comment by eregon** [comment]
Fixed in 9cda04fc0e18f8983e7990914dfac96b6a45b3f8 along quite a few other public methods.
Thanks for the report!
